### DISCUSSION TITLE
Support for `useInfiniteQueries`?

### DISCUSSION THREAD

**Discussion by Brodzko** [TanStack/query#3016] [Ideas]
Hello!

I'm curious whether support for `useInfiniteQueries` - similar to already existing `useQueries` is planned? My use-case is a nested list of items, where each level is paginated by the API and should be lazy loaded on scroll.

If there already is a canonical way to go about this I'm happy to learn :)

Many thanks!

**Comment by TkDodo** [comment]
I can't say it's on the list, no. But when I look at `usInfiniteQuery` - it's basically the same as `useQuery`, as they both `useBaseQuery` under the hood, just with a different observer - `InfiniteQueryObserver`. So maybe the `QueriesObserver` is extensible in the same way 🤔 . Do you maybe want to take a look and tackle it?

**Reply by sventschui** [reply]
I just took a stab at this in a private repo and from what I can see this should work out:

- Ensure `InfiniteQueryObserver.setOptions` allows to pass `notifyOptions` to its super class/method
- Make the `QueriesObserver` class generic to make it work with a list of `QueryObserver` or a list of `InfiniteQueryObserver` (maybe introduce an `AbstractQueriesObserver` that gets the `QueryObserver` or `InfiniteQueryObserver` class as a constructor argument and subclass it to `QueriesObserver` and `InfinitQueriesObserver`) 
- Extract `useQueries` into `useQueriesBase` and use it in `useQuery` and `useInfiniteQueries`

Not sure if I find time to contribute this upstream as project deadlines are near. Happy if anyone wants to follow this approach before I get to contribute this.
